AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java

```diff
@@ -190,14 +197,32 @@ public void afterTextChanged(final Editable editable) {
                 setupEnter.setTextColor(ContextCompat.getColor(this, R.color.colorAccent));
             } else {
                 setupEnter.setTextColor(ContextCompat.getColor(this, R.color.textbox_secondary));
+                final InputMethodManager inputMethodManager =
+                        (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
+                inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
             }
         });
 
         joinButton = findViewById(R.id.setup_button);
         joinButtonText = findViewById(R.id.setup_button_text);
+
+        final ToggleButton deviceOptionsButton = findViewById(R.id.device);
+        deviceOptionsButton.setOnFocusChangeListener((view, hasFocus) -> {
+            if (hasFocus) {
+                openAudioDeviceList();
+            }
+        });
+        deviceOptionsButton.setOnClickListener(l -> {
+            openAudioDeviceList();
+        });
+
         hidePermissionsWarning();
     }
 
+    private void openAudioDeviceList() {
+        new AudioDeviceSelectionPopupWindow(this, audioSessionManager);
```

Comment:
Instead of constructing `AudioDeviceSelectionPopupWindow` every time, we can instantiate it once and expose a `show()` method.

Reply:
Updated the logic to not create popup window every time.